controller/src/main/java/io/pravega/controller/server/security/auth/LegacyAuthorizationResourceImpl.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.security.auth;
+
+import io.pravega.common.Exceptions;
+
+/**
+ * A legacy implementation that constructs resource strings in the old format.
+ */
+public final class LegacyAuthorizationResourceImpl implements AuthorizationResource {
```

Comment:
I suppose we are deprecating the previous format, how are we going to indicate that it is deprecated? Via documentation? We can certainly annotate this class, but that's not visible to users.

Reply:
Actually, we are removing support for the old format altogether. This class is there only so that it is easy to see what was the old format and compare it with the new format.

Reply:
I wanted to keep it at least for some time so that its easier to inspect the differences and compare the two at development time. I shall remove them once this change has been well tested through the normal course of testing.

Reply:
In the latest comment that I have just pushed, I've added a deprecation note in the class JavaDoc. (I couldn't use the deprecation annotation, as the current build configuration fails build if classes decorated with that annotation are used.) 

Also, these classes are used internally for representing resources and authorization and aren't meant to be used by external parties.

Reply:
I'm not sure this makes sense. If we need to revert or anything, the we can go back to a previous version, that's a reason for using version control. If this code is only being kept for reference, then let's please remove it. If it is being used anywhere, then keep it.

Reply:
Ok. I have removed the two classes that show legacy logic in the latest commit that I just pushed.